File: Library/Server.Database/Models/Product.cs
using System.Text.Json.Serialization;

namespace Server.Database.Models;

public class Product
{
    public long Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Size { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; }
    public DateTime? DeletedAt { get; set; }
    
    [JsonIgnore]
    public long? GroupId { get; set; }
    [JsonIgnore]
    public long? FileId { get; set; }

    public ProductGroup? Group { get; set; }
    public File? File { get; set; }
    public List<ProductType> Types { get; set; } = new();
    public List<ProductDescription> Descriptions { get; set; } = new();
}

File: Client/react/src/lib/models/Product.ts
import {ProductType} from "./ProductType";
import {ProductDescription} from "./ProductDescription";
import {ProductGroup} from "./ProductGroup";
import {FileReaderDto} from "./FileReaderDto";
import {FileEntity} from "./File";

export interface Product {
    id?: number;
    name: string;
    size: string;
    file?: FileEntity;
    fileDto?: FileReaderDto;
    group?: ProductGroup;
    types?: ProductType[];
    descriptions?: ProductDescription[];
}